### ISSUE TITLE
Round Path URL

### ISSUE THREAD

**Issue by royalterra** [idno/Twitter#45]
Possible that is the problem of some issues here. If you install IDNo in a folder the twitter plugin is the only in the user configuration the link twitter point to the root domain and the Bookmarklet also pint to the root domain and in consequence no works.
Please correct detect path of installation in the user side.
Post is cross posting with no issues anyway to twitter but url

**Comment by kylewm** [comment]
Hi @royalterra, thanks for filing this issue. I'm having some trouble understanding the description, so let me rephrase and you can tell me if I'm right.

You've installed Known in a subdirectory so it's at (for example) `http://example.com/blog` instead of `http://example.com/`.

When you try to reply or share a URL with the bookmarklet, it opens at the root `http://example.com/share?share_url=...` which of course doesn't work.

And when you syndicate to Twitter, it includes a link to the original post in the tweet, but the link is to wrong too, like `http://example.com/2016/blah-blah-blah`

Is that right so far?

**Comment by royalterra** [comment]
`You've installed Known in a subdirectory so it's at (for example) http://example.com/blog instead of http://example.com/. **Yes that is correct**

`When you try to reply or share a URL with the bookmarklet, it opens at the root http://example.com/share?share_url=... which of course doesn't work.` Yes qhen I try to share using Firefox Services go to the root domain

`And when you syndicate to Twitter, it includes a link to the original post in the tweet, but the link is wrong too, like http://example.com/2016/blah-blah-blah` No That is incorrect if I share to twitter the link is correct. to the subfolder 

The link is incorrect in user section if you click in user configuration and you go to Twitter go to http://example.domain/account/twitter/ and need to go to http://example.domain/blog/account/twitter/ 
If I change the URL in the browser open but the direct link is round 

The seam issues are for the plugins     Foursquare,  Diigo and WordPress any other plugins work perfect if the root path is different .






`

**Comment by kylewm** [comment]
Sent a PR to fix the issue with the /account/twitter/ link.

The Firefox plugin already [looks correct](https://github.com/idno/Known/blob/master/IdnoPlugins/Firefox/templates/default/account/firefox/manifest.tpl.php#L13) to me, so I'm not quite sure what could be going on there. Perhaps try reinstalling the FIrefox service.

**Comment by royalterra** [comment]
No is not only a issue of Firefox is a issue of Internal Links 
I go to https://mydomain.com/know
and all work fine apparently after I go to https://mydomain.com/known/account/settings/ 
And all look OK 

**NOW I GO TO TWITTER and need to open https://mydomain.com/known/account/twitter/ but no go to that URL . The link go to https://mydomain.com/account/twitter/ what is totally round . **

**Comment by royalterra** [comment]
You can see at the end of the record in table **entities** the url STATUS redirect to the root DOMAIN
`{"access":"PUBLIC","owner":"https:\/\/Mydomain.com\/shout\/profile\/anonimo","body":"Hello . For all my Followers in Facebook and Twitter","inreplyto":false,"tags":false,"slug":"hello-for-all-my-followers-in-facebook-and-twitter","created":1461460457,"updated":1461460458,"_id":"d50280f69219cf87b60791de7cb32025","uuid":"https:\/\/Mydomain.com\/shout\/2016\/hello-tov-for-all-my-followers-in-facebook-and-twitter","syndicatedto":null,"shorturl":"5yzDM","posse":{"twitter":[{"url":"https:\/\/twitter.com\/Mydomain_com\/status\/724043755657920512","identifier":"@Mydomain_com","item_id":"724043755657920512","account_id":"Mydomain_com"}]},"entity_subtype":"IdnoPlugins\\Status\\Status"}`

**Comment by kylewm** [comment]
> *NOW I GO TO TWITTER and need to open https://mydomain.com/known/account/twitter/ but no go to that URL . The link go to https://mydomain.com/account/twitter/ what is totally round . *

Got it, that is what the PR I sent above fixes. You can see it's a pretty trivial change, but it has to be done for each plugin individually, which is a bit of a pain.

> `"uuid":"https:\/\/Mydomain.com\/shout\/2016\/hello-tov-for-all-my-followers-in-facebook-and-twitter"`

This is the entry that matters -- and it appears to have the correct subdirectory `shout`

**Comment by royalterra** [comment]
Hi Thanks I will wait they include your pull request and try again

**Comment by dosch** [comment]
I second the described behavior. Looking forward to this update.
